src/ResolveAliasesTransformer.cpp

```diff
@@ -343,8 +342,8 @@ std::unique_ptr<AstClause> ResolveAliasesTransformer::removeTrivialEquality(cons
     // add all literals, except filtering out t = t constraints
     for (AstLiteral* literal : clause.getBodyLiterals()) {
         if (auto* constraint = dynamic_cast<AstBinaryConstraint*>(literal)) {
-            // don't filter out `FEQ` constraints, since `x = x` can fail when `x` is a NaN
-            if (constraint->getOperator() == BinaryConstraintOp::EQ) {
+            // TODO: don't filter out `FEQ` constraints, since `x = x` can fail when `x` is a NaN
+            if (isEqConstraint(constraint->getOperator())) {
```

Comment:
This isn't correct for `float`/`FEQ` and should cause `comparison.dl` to fail.

Reply:
It seems that without adding `FEQ` here, the example above (and the `semantic/aggregate5`) won't terminate. What is `comparison.dl`?

Reply:
Ok, I've found it. I think the test isn't declared as part of the testsuite and it is not running right now (it should be added to `semantic.at`).

Reply:
Indeed the test fails on this branch, but it also fails when run on the master [(link to Travis)](https://travis-ci.org/github/darth-tytus/souffle/builds/663830941)

Reply:
Oops. Sorry let me go take a look.

Reply:
```
+++ /home/travis/build/darth-tytus/souffle/tests/testsuite.dir/at-groups/115/stdout	2020-03-18 08:01:12.113035459 +0000
@@ -0,0 +1,4 @@
+1a2
+>  =	-0	0
+5a7
+>  =	0	-0
./semantic.at:102: exit code was 1, expected 0
```
I cannot account for why whose two lines are missing from the expected output. I'll make a PR to add the test case to the suite and fix the expected output.